dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/bundle/ReportSummaryDeleteIntegrationTest.java

```diff
@@ -75,12 +78,36 @@ protected void initTest() throws IOException {
     assertImportedObjects(13, persistenceReport, TrackerType.TRACKED_ENTITY);
     assertImportedObjects(2, persistenceReport, TrackerType.ENROLLMENT);
     assertImportedObjects(2, persistenceReport, TrackerType.EVENT);
+    assertImportedObjects(2, persistenceReport, TrackerType.RELATIONSHIP);
     assertEquals(6, manager.getAll(Enrollment.class).size());
     assertEquals(
         persistenceReport.getTypeReportMap().get(TrackerType.EVENT).getStats().getCreated(),
         manager.getAll(Event.class).size());
   }
 
+  @Test
+  void shouldFailToDeleteNotExistentRelationship() throws IOException {
+    TrackerObjects trackerObjects = fromJson("tracker/relationship_not_existent_for_deletion.json");
+    TrackerImportParams params = new TrackerImportParams();
+    params.setImportStrategy(TrackerImportStrategy.DELETE);
+    assertEquals(1, trackerObjects.getRelationships().size());
+
+    ImportReport importReport = trackerImportService.importTracker(params, trackerObjects);
+    assertHasOnlyErrors(importReport, E4016);
+  }
+
+  @Test
+  void shouldSuccessfullyDeleteRelationships() throws IOException {
```

Comment:
Is there a test for the exceptional case you mention in the PR description i.e. unidirectional relationships where the user does not have access to from? That seems important to have 😅

Reply:
Done